app/models/secret_storage.rb

```diff
@@ -124,5 +151,19 @@ def read(key, include_secret: false)
     def backend
       BACKEND
     end
+
+    def generate_secret_key(environment=nil, project=nil, deploy_group=nil, key=nil)
+      if environment.nil? || project.nil? || deploy_group.nil? || key.nil?
+        raise ArgumentError.new("missing paramater")
+      end
+      environment.to_s + "/" + project.to_s + "/" + deploy_group.to_s + "/" + key.to_s
+    end
+
+    def parse_secret_key(key, field)
+      return false if key.nil?
+      index = SecretStorage::SECRET_KEYS.index(field)
+      value = key.split('/', SecretStorage::SECRET_KEYS.count)[index]
+      value.nil? ? false : value
```

Comment:
this line could also go away ... no read need to return false when nil is good enough ...